kong/router/atc_compat.lua

```diff
@@ -428,12 +429,93 @@ function _M.new(routes, cache, cache_neg)
       routes = routes_t,
       services = services_t,
       fields = inst:get_fields(),
-      cache = cache,
-      cache_neg = cache_neg,
     }, _MT)
 end
 
 
+local function is_route_changed(a, b)
+  return a.updated_at ~= b.updated_at
+end
+
+
+local function new_from_previous(routes, is_traditional_compatible, pre_router)
+  local inst = pre_router.router
+  local pre_routes = pre_router.routes
+  local pre_services = pre_router.services
+
+  -- create or update routes
+  for _, r in ipairs(routes) do
+    local route = r.route
+    local route_id = route.id
+
+    if not route_id then
+      return nil, "could not categorize route"
+    end
+
+    route.touched = true
```

Comment:
```suggestion
    route.seen = true
```

Probably a better name.